app/policies/comment_policy.rb

```diff
@@ -27,6 +27,10 @@ def preview?
     true
   end
 
+  def moderator_create?
+    !user_is_banned? && !user_is_comment_banned? && !user_is_blocked? && (user_is_moderator? || minimal_admin?)
```

Comment:
Do we let users keep moderator tags when they are banned? This method in the response template policy class doesn't include the banned checks, I'm wondering if we need them here. 
```ruby
def moderator_create?
  user_is_moderator? && record_is_mod_comment?
end
```